.github/workflows/build.yml

```diff
@@ -4,16 +4,27 @@ on:
   push:
     branches:
       - master
+
+    paths-ignore:
+      - '**.md'
+      - '.github/**'
+      - '!.github/workflows/build-*'
+      
+
   pull_request:
     types: [opened, synchronize, reopened, ready_for_review]
   workflow_dispatch:
 
-jobs:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
 
+jobs:
   build:
     name: Build
     runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
```

Comment:
Why do we not want to fail fast? In the case where we have a flaky test, are we expecting to let the workflow run on both ubuntu and macos all the way through?

Reply:
Yes. Based on these reasons though:
1. Fail-fast is not good for retry workflow since it will disapprove the success jobs. 
2. Even if we have a flaky test, it does not mean that it would fail based on the OS. Therefore, it should show the corresponding result with corresponding OS. Moreover, it would reduce the attempt to address the flakey test with the only failed OS.